### ISSUE TITLE
Enhancement: padding-line-between-statements support enum

### ISSUE THREAD

**Issue by ddrozdov** [typescript-eslint/typescript-eslint#8240]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/padding-line-between-statements/

### Description

Add `enum` to supported statement types.

### Fail

```TypeScript
const str = "bla";
enum First {
  ONE = "one"
}
enum Second {
  TWO = "two"
}
```

### Pass

```TypeScript
const str = "bla";

enum First {
  ONE = "one"
}

enum Second {
  TWO = "two"
}
```

### Additional Info

Easily resolved by adding this line:
```
enum: newKeywordTester(utils_1.AST_NODE_TYPES.TSEnumDeclaration, 'enum'),
```

**Comment by auvred** [comment]
All stylistic rules are deprecated in typescript-eslint. They are now maintained by [ESLint Stylistic](https://eslint.style/) project. You can file an issue in the eslint-stylistic repo: https://github.com/eslint-stylistic/eslint-stylistic/issues

Ref: https://typescript-eslint.io/blog/deprecating-formatting-rules/

**Comment by ddrozdov** [comment]
https://github.com/eslint-stylistic/eslint-stylistic/issues/263
